### ISSUE TITLE
Compiler warning in Dart 2.1.0 when verifying void returning methods

### ISSUE THREAD

**Issue by v3rm0n** [dart-lang/mockito#163]
Looks like Dart 2.1.0 has a change that generates a compiler warning when you are using a void returning expression inside a method call https://github.com/dart-lang/sdk/commit/9eac90d2a257c68fd1c24a65cc2d1fc3018375e0

So
```
verify(cat.sleep());
```
Will cause
```
compiler message: test/src/cat_test.dart:94:35: Warning: This expression has type 'void' and can't be used.
compiler message:         verify(cat.sleep());
compiler message:                    ^
```

**Comment by MichaelRFairhurst** [comment]
Thanks for filing! This is a known SDK bug: https://github.com/dart-lang/sdk/issues/30470#issuecomment-412581103

**Comment by MichaelRFairhurst** [comment]
You can work around this warning by casting to dynamic (which is free):

```
verify(cat.sleep() as dynamic);
```

But you should not have to do this once the bug is fixed!

**Comment by srawlins** [comment]
Dart 2.1.0 is not released yet. Which 2.1.0-dev version is this?

**Comment by MichaelRFairhurst** [comment]
I think all of them

**Comment by kmillikin** [comment]
This should be fixed by 9bdf2489b54841567005eaff7033f3451a047b37.  How can we verify that?

**Comment by srawlins** [comment]
I think this has been fixed. We haven't bumped into this in over a year.
